internal/cli/watch_opts.go

```diff
@@ -55,6 +67,27 @@ func (opts *WatchOpts) Watch(f Watcher) error {
 	}
 }
 
+func (opts *WatchOpts) exponentialBackoff(err error) bool {
+	if opts.n <= 8 && checkForError(err, "CLUSTER_NOT_FOUND") {
+		backoff := math.Pow(base, float64(opts.n))
+		opts.n *= 2
+		sleepTime := time.Duration(backoff) * time.Second
+		time.Sleep(sleepTime)
+		return true
+	}
+	return false
+}
+
+func checkForError(err error, code string) bool {
+	var atlasErr *atlas.ErrorResponse
+	if errors.As(err, &atlasErr) {
+		if atlasErr.ErrorCode == code {
+			return true
+		}
+	}
+	return false
```

Comment:
```suggestion
	return errors.As(err, &atlasErr) && atlasErr.ErrorCode == code 
```

Reply:
also consider, given this is a oneliner if it really deserves a dedicated method

Reply:
Thanks, looks cleaner. I removed the method and moved the variables and logic to the `exponentialBackoff`.